File: backend/app/schemas/analyze.py
"""Schemas for POST /api/v1/analyze (orchestrator — implemented CP15)."""

from pydantic import BaseModel

from app.schemas.common import ArticleTextRequest
from app.schemas.entities import EntitiesResponse
from app.schemas.keywords import KeywordsResponse
from app.schemas.sentiment import SentimentResponse
from app.schemas.summarize import SummarizeResponse


class AnalyzeRequest(ArticleTextRequest):
    """Full analysis in one request for the interactive UI (O6)."""


class AnalyzeResponse(BaseModel):
    summary: SummarizeResponse
    sentiment: SentimentResponse
    keywords: KeywordsResponse
    entities: EntitiesResponse

File: frontend/src/types/analyze.ts
/** Mirror of backend/app/schemas/analyze.py (CP15) */

import type { ArticleTextRequest } from "@/types/common";
import type { EntitiesResponse } from "@/types/entities";
import type { KeywordsResponse } from "@/types/keywords";
import type { SentimentResponse } from "@/types/sentiment";
import type { SummarizeResponse } from "@/types/summarize";

export type AnalyzeRequest = ArticleTextRequest;

export type AnalyzeResponse = {
  summary: SummarizeResponse;
  sentiment: SentimentResponse;
  keywords: KeywordsResponse;
  entities: EntitiesResponse;
};
